packages/compass-crud/src/stores/crud-store.ts

```diff
@@ -706,19 +747,25 @@ class CrudStoreImpl
         'replace'
       );
       if (error) {
-        doc.emit('update-error', error.message);
+        doc.onUpdateError(error as Error);
       } else {
-        doc.emit('update-success', d);
+        doc.onUpdateSuccess(d);
         const index = this.findDocumentIndex(doc);
-        this.state.docs![index] = new HadronDocument(d);
-        this.trigger(this.state);
+        const newDocs = this.state.docs
+          ? [...this.state.docs]
+          : this.state.docs;
+        newDocs?.splice(index, 1, new HadronDocument(d));
+        this.setState({
+          docs: newDocs,
+        });
```

Comment:
I removed `this.trigger(this.state)` assuming that `this.setState` will do the same thing. Let me know if that's not the case.